[Software Delivery] Ruby library documentation improvements after manual API release (#20898)
DataDog/documentation: content/en/continuous_integration/tests/setup/ruby.md

### Adding custom metrics to tests

> <div class="alert alert-info">`Datadog::CI` public API is available in `ddtrace` gem versions >= 1.17.0</div>

Reviewer: ```suggestion
<div class="alert alert-info">The <code>Datadog::CI</code> public API is available in <code>ddtrace</code> gem versions >= 1.17.0</div>
```

All other [Datadog Tracer configuration][6] options can also be used.

> ## Using additional instrumentations

Reviewer: ```suggestion
## Using additional instrumentation
```

> It can be useful to have rich tracing information about your tests that includes time spent performing database operations
> or other external calls like here:

Reviewer: ```suggestion
It can be useful to have rich tracing information about your tests that includes time spent performing database operations or other external calls, as seen in the following flame graph:
```

> {{< img src="continuous_integration/tests/setup/ci-ruby-test-trace-with-redis.png" alt="Test trace with redis instrumented" >}}

Reviewer: ```suggestion
{{< img src="continuous_integration/tests/setup/ci-ruby-test-trace-with-redis.png" alt="Test trace with Redis instrumented" >}}
```

> In order to achieve this you can configure additional instrumentations in your configure block:

Reviewer: ```suggestion
To achieve this, configure additional instrumentation in your `configure` block:
```

```ruby
if ENV["DD_ENV"] == "ci"
  Datadog.configure do |c|
    #  ... ci configs and instrumentation here ...
    c.tracing.instrument :redis
    c.tracing.instrument :pg
    # ... any other instrumentations supported by ddtrace gem ...
  end
end
```

> ...or enable auto instrumentation in your test_helper/spec_helper:

Reviewer: ```suggestion
Alternatively, you can enable automatic instrumentation in `test_helper/spec_helper`:
```

```ruby
require "ddtrace/auto_instrument" if ENV["DD_ENV"] == "ci"
```

> Note: in CI mode these traces are submitted to CI Visibility, they do **not** show up in Datadog APM.

Reviewer: ```suggestion
**Note**: In CI mode, these traces are submitted to CI Visibility, and they do **not** show up in Datadog APM.
```

> For the full list of available instrumentations see [ddtrace documentation][8]

Reviewer: ```suggestion
For the full list of available instrumentation methods, see the [`ddtrace` documentation][8]
```
